diff --color -rupN org/libvo/cgx_common.c new/libvo/cgx_common.c
--- org/libvo/cgx_common.c	2023-08-09 09:40:55.631578076 +0200
+++ new/libvo/cgx_common.c	2023-08-09 11:01:25.488034263 +0200
@@ -697,14 +697,16 @@ bool is_uniconified()
 
 bool enable_Iconify()
 {
-	const char *files[]={"progdir:MPlayer","progdir:MPlayer-AltiVec","progdir:mplayer","progdir:mplayer-altivec","progdir:mplayer-nonaltivec","envarc:sys/def_video", "envarc:sys/def_tool",NULL};
+/*	const char *files[]={"progdir:MPlayer","progdir:MPlayer-AltiVec","progdir:mplayer","progdir:mplayer-altivec","progdir:mplayer-nonaltivec","envarc:sys/def_video", "envarc:sys/def_tool",NULL};
 	int n;
 
 	for (n=0;files[n];n++)
 	{
 		_dobj = GetDiskObjectNew( files[n] );
 		if (_dobj) break;
-	}
+	}*/
+	_dobj = GetDiskObject( fullpath_mplayer );
+	if(!_dobj) _dobj = GetDiskObject("ENVARC:Sys/def_shell");
 
 	if (_dobj)
 	{
