--- SDH6:mplayer-src-10082023/libao2/ao_sdl.c	2023-09-18 14:53:35 
+++ libao2/ao_sdl.c	2023-09-18 14:46:18 
@@ -315,5 +315,5 @@ static int play(void* data,int len,int f
 // return: delay in seconds between first and last sample in buffer
 static float get_delay(void){
     int buffered = av_fifo_size(buffer); // could be less
-    return (float)(buffered + ao_data.buffersize)/(float)ao_data.bps;
+    return (float)(buffered + abs(ao_data.buffersize))/(float)ao_data.bps;
 }
